### PR TITLE
Remove ConfigurationSetting parameter from getSetting and deleteSetting methods

### PR DESCRIPTION
**Pull request by mssfang**
Removed ConfigurationSetting parameter from getSetting and deleteSetting method parameter. Use the direct configuration elements directly instead.

fixes Change getSetting and deleteSetting APIs #5371

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 1]
```diff
@@ -400,8 +402,8 @@
      * @throws HttpResponseException If the {@code} key is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<ConfigurationSetting> getSetting(ConfigurationSetting setting) {
-        return withContext(context -> getSetting(setting, context))
+    public Mono<ConfigurationSetting> getSetting(String key, String label, String dateTime) {
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: DateTime as a string? I would be hoping for an Instant. Having it as a string brings up many questions... like "What format should it be? What locale do I parse this string in?"

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 2]
```diff
@@ -517,19 +525,23 @@
...
-        return withContext(context -> deleteSetting(setting, context));
+    public Mono<Response<ConfigurationSetting>> deleteSettingWithResponse(String key, String label, String etag) {
+        return withContext(context -> deleteSetting(key, label, etag, context));
     }
 
-    Mono<Response<ConfigurationSetting>> deleteSetting(ConfigurationSetting setting, Context context) {
+    Mono<Response<ConfigurationSetting>> deleteSetting(String key, String label, String etag, Context context) {
         // Validate that setting and key is not null. The key is used in the service URL so it cannot be null.
-        validateSetting(setting);
+        if (key == null) {
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Based on the spec, this should be Objects.requireNotNull and throw NullPointerException.
https://azure.github.io/azure-sdk/java_implementation.html

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Agreed, have this take an `OffsetDateTime` and we will handle the string conversion

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 3]
```diff
@@ -391,7 +392,9 @@
      *
      * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.getSetting#ConfigurationSetting}
      *
-     * @param setting The setting to retrieve based on its key and optional label combination.
+     * @param key The key of the setting to retrieve based on its key and optional label combination.
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Given that we are splaying out the parameters we should change the wording here.

`The key of the setting to retrieve`
`Optional. A label used to create the key-label combination to identify the setting`.
`Optional. A date time used to retrieve the setting state at the passed time.`

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 4]
```diff
@@ -424,19 +429,19 @@
      * @throws HttpResponseException If the {@code} key is an empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<ConfigurationSetting>> getSettingWithResponse(ConfigurationSetting setting) {
-        return withContext(context -> getSetting(setting, context));
+    public Mono<Response<ConfigurationSetting>> getSettingWithResponse(String key, String label, OffsetDateTime dateTime) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Should this and the other method that takes the same parameters be merged into a single API method?

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 5]
```diff
@@ -475,28 +480,30 @@
      *
      * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.deleteSetting#ConfigurationSetting}
      *
-     * @param setting The ConfigurationSetting to delete.
+     * @param key The key of the setting to delete.
+     * @param label The label of the setting to delete.
+     * @param eTag The entity tag of the setting to delete.
      * @return The deleted ConfigurationSetting or {@code null} if didn't exist. {@code null} is also returned if the
      * {@code key} is an invalid value or {@link ConfigurationSetting#getETag() etag} is set but does not match the
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: This should say `the passed eTag does not match the current eTag`, we aren't pulling the value out of the `ConfigurationSetting` anymore.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 6]
```diff
@@ -505,31 +512,34 @@
...
-        validateSetting(setting);
-
-        return service.delete(serviceEndpoint, setting.getKey(), setting.getLabel(), getETagValue(setting.getETag()),
+        Objects.requireNonNull(key);
+        return service.delete(serviceEndpoint, key, label, getETagValue(eTag),
             null, context)
-            .doOnRequest(ignoredValue -> logger.info("Deleting ConfigurationSetting - {}", setting))
-            .doOnSuccess(response -> logger.info("Deleted ConfigurationSetting - {}", response.getValue()))
-            .doOnError(error -> logger.warning("Failed to delete ConfigurationSetting - {}", setting, error));
+            .doOnRequest(ignoredValue -> logger.info("Deleting ConfigurationSetting - key:{}, label:{}, etag:{}",
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: Let's add a space after each `:`

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java [hunk 7]
```diff
@@ -435,12 +446,12 @@ public ConfigurationSetting deleteSetting(ConfigurationSetting setting) {
...
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Response<ConfigurationSetting> deleteSettingWithResponse(ConfigurationSetting setting, Context context) {
-        return deleteSetting(setting, context);
+    public Response<ConfigurationSetting> deleteSettingWithResponse(String key, String label, String etag, Context context) {
+        return deleteSetting(key, label, etag, context);
     }
 
-    private Response<ConfigurationSetting> deleteSetting(ConfigurationSetting setting, Context context) {
-        return client.deleteSetting(setting, context).block();
+    private Response<ConfigurationSetting> deleteSetting(String key, String label, String etag, Context context) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: Delete this method and call the async client in the maximal overload.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java [hunk 8]
```diff
@@ -351,12 +357,13 @@ public ConfigurationSetting getSetting(ConfigurationSetting setting) {
...
-    public Response<ConfigurationSetting> getSettingWithResponse(ConfigurationSetting setting, Context context) {
-        return getSetting(setting, context);
+    public Response<ConfigurationSetting> getSettingWithResponse(String key, String label, OffsetDateTime dateTime,
+                                                                 Context context) {
+        return getSetting(key, label, dateTime, context);
     }
 
-    private Response<ConfigurationSetting> getSetting(ConfigurationSetting setting, Context context) {
-        return client.getSetting(setting, context).block();
+    private Response<ConfigurationSetting> getSetting(String key, String label, OffsetDateTime dateTime, Context context) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClient.java]: Delete this method as well

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java (hunk 3 above)
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]: The Javadoc here is weird. Also, the NullPointerException in throws is wrong.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java [hunk 9]
```diff
@@ -475,28 +480,30 @@
...
+    public Mono<ConfigurationSetting> deleteSetting(String key, String label, String eTag) {
+        return withContext(context -> deleteSetting(key, label, eTag, context))
             .flatMap(response -> Mono.justOrEmpty(response.getValue()));
     }
 
     /**
-     * Deletes the {@link ConfigurationSetting} with a matching key, along with the given label and etag.
+     * Deletes the {@link ConfigurationSetting} with a matching key, along with the given label and eTag.
      *
      * If {@link ConfigurationSetting#getETag() etag} is specified and is not the wildcard character ({@code "*"}), then
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]:
```suggestion
      * If {@link ConfigurationSetting#getETag() ETag} is specified and is not the wildcard character ({@code "*"}), then
```

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java (hunk 4 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java]:
```java
  public Mono<Response<ConfigurationSetting>> getSettingWithResponse(String key, String label, OffsetDateTime dateTime) 
  public Mono<ConfigurationSetting> getSetting(String key, String label, OffsetDateTime dateTime)
```
Is that the methods you are refering to? If yes, those two methods are different used. With Response return type has extra infornation than just value.

**Comment by joshfree** [comment]
Closing PR based on offline conversation.  These changes are incorporated in later PRs

**Pull request closed without merging**